apis/python/tests/test_dataframe.py

```diff
@@ -147,6 +147,36 @@ def test_dataframe(tmp_path, arrow_schema):
         assert isinstance(A._handle._handle, soma.pytiledbsoma.SOMADataFrame)
 
 
+def test_dataframe_reopen(tmp_path, arrow_schema):
+    soma.DataFrame.create(
+        tmp_path.as_posix(), schema=arrow_schema(), tiledb_timestamp=1
+    )
+
+    with soma.DataFrame.open(tmp_path.as_posix(), "r") as sdf1:
+        with raises_no_typeguard(ValueError):
+            sdf1.reopen("invalid")
+
+        with sdf1.reopen("w") as sdf2:
+            with sdf2.reopen("r") as sdf3:
+                assert sdf1.mode == "r"
+                assert sdf2.mode == "w"
+                assert sdf3.mode == "r"
+
+    with soma.DataFrame.open(tmp_path.as_posix(), "r") as sdf1:
+        with sdf1.reopen("r") as sdf2:
+            assert sdf1.mode == "r"
+            assert sdf2.mode == "r"
+
+    with soma.DataFrame.open(tmp_path.as_posix(), "w") as sdf1:
+        with sdf1.reopen("w") as sdf2:
+            assert sdf1.mode == "w"
+            assert sdf2.mode == "w"
+
+    with soma.DataFrame.open(tmp_path.as_posix(), "w", tiledb_timestamp=1) as sdf1:
+        with sdf1.reopen("r") as sdf2:
+            assert sdf1.tiledb_timestamp < sdf2.tiledb_timestamp
```

Comment:
Just for completeness, it wouldn't be a bad idea to copy this assertion to the above tests, and ensure no interaction between various mode changes and the timestamp.